src/libcmd/repl.cc

```diff
@@ -506,11 +506,15 @@ ProcessLineResult NixRepl::processLine(std::string line)
         auto editor = args.front();
         args.pop_front();
 
+        // avoid garbling the editor with the progress bar
+        logger->pause();
```

Comment:
```suggestion
        logger->pause();
        Finally resume([&]() { logger->resume(); });
```
This should ensure that the logger restarts even if the editor throws an exception